node/src/accountant/scanners/mod.rs

```diff
@@ -942,6 +944,8 @@ impl ReceivableScanner {
                     }
                     Err(e) => panic!("Commit of received transactions failed: {:?}", e),
                 }
+            } else {
+                warning!(logger, "The new_start_block ({}) is less than the current_start_block ({}). This is not a problem but by checking we avoid rescanning the same blocks again later.", &new_start_block, &current_start_block);
```

Comment:
If I remember correctly, there are plans that the start_block parameter will be eventually hidden from the user. So I like what you did here and if not best then it will go away and solve this tricky caveat on its own. :-)